cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -1113,6 +1113,13 @@ private module GetConvertedResultExpression {
       result = tas.getExtent().getExpr() and
       instr = tas.getInstruction(any(AllocationExtentConvertTag tag))
     )
+    or
```

Comment:
Is there a reason why we don't make this change at the IR level to have the instruction also return `ParenthesisExpr` rather than add a special case here? The other cases feel more like adjusting to make dataflow easier rather than more correct but this just feels like a missing case for at the IR level.

There's not really any reason to change it here I am just wanting to understand the decision.

Reply:
I think there are two reasons that combine to fully explain this:
1. Given an instruction `i` there's always at most one result for `instr.getConvertedResultExpression()`.
2. Reading https://github.com/github/codeql/blob/main/cpp/ql/lib/semmle/code/cpp/ir/implementation/raw/internal/TranslatedExpr.qll#L3236 it sounds like it's a conscious decision to not generate a `CopyValue` instruction for parenthesis expressions.

So since `instr.getConvertedResultExpression()` can only return one expression, and there's no instruction generated for parenthesis expressions, I don't think there's an easy way to fit this change into the IR level.